packages/flutter/test/rendering/object_test.dart

```diff
@@ -246,6 +246,51 @@ void main() {
     renderObject.dispose();
     expect(renderObject.debugLayer, null);
   });
+
+  test('PaintingContext.pushClip* checks that Clip.none means a null approximate clip', () {
+    int paintCount = 0;
+    void paint(PaintingContext context, Offset offset) {
+      paintCount += 1;
+    }
+
+    final RenderSizedBox child = RenderSizedBox(const Size(300, 300));
+
+    final TestBadClipRenderObject object = TestBadClipRenderObject(child, (PaintingContext context) {
+      paintCount += 1;
+      expect(
+        () => context.pushClipRect(false, Offset.zero, Rect.zero, paint, clipBehavior: Clip.none),
+        throwsFlutterError,
+      );
+      expect(
+        () => context.pushClipRRect(false, Offset.zero, Rect.largest, RRect.zero, paint, clipBehavior: Clip.none),
+        throwsFlutterError,
+      );
+      expect(
+        () => context.pushClipPath(false, Offset.zero, Rect.largest, Path(), paint, clipBehavior: Clip.none),
+        throwsFlutterError,
+      );
+    });
+    layout(object);
+    object.owner!.flushCompositingBits();
+    PaintingContext(ContainerLayer(), Rect.largest).paintChild(object, Offset.zero);
+
+    expect(paintCount, 1);
+  });
+
+  test('PaintingContext.pushClip* allows a RO clip children with none/hardEdge', () {
+    final RenderSizedBox child = RenderSizedBox(const Size(300, 300));
+    final RenderSizedBox specialChild = RenderSizedBox(const Size(200, 200));
+
+
+    final TestMultiClipBehaviorClipRenderObject object = TestMultiClipBehaviorClipRenderObject(child, specialChild);
+
+    layout(object);
+    object.owner!.flushCompositingBits();
+    expect(
+      () => PaintingContext(ContainerLayer(), Rect.largest).paintChild(object, Offset.zero),
+      returnsNormally,
```

Comment:
@goderbauer specifically the RO I added below does what I think you were asking for, and makes sure to return the right response from `describeApproximatePaintClip` based on which child you ask for.